java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/diff/ObfDiffMerger.java

```diff
@@ -145,8 +163,22 @@ private void mergeRelationDiff(String[] args) throws IOException {
 		deleteId = ri.searchRouteEncodingRule(OSMAND_CHANGE_TAG, OSMAND_CHANGE_VALUE);
 		TLongObjectHashMap<RouteDataObject> relRouteData = relObf.getRoutingData();
 		TLongObjectHashMap<RouteDataObject> commonRouteData = commonObf.getRoutingData();
+
+		deletedShiftedIds.clear();
+		if (deleteId > 0) {
+			for (Long id : commonRouteData.keys()) {
+				RouteDataObject commonObj = commonRouteData.get(id);
+				if (commonObj.containsType(deleteId)) {
+					deletedShiftedIds.add(commonObj.getId() >> SHIFT_ID);
+				}
+			}
+		}
+
 		cnt = 0;
 		for (Long id : relRouteData.keys()) {
+			if (deletedShiftedIds.contains(id >> SHIFT_ID)) {
+				continue;
+			}
```

Comment:
Below there is a check 
```
	for (Long id : relMapData.keys()) {
				BinaryMapDataObject relObj = relMapData.get(id);
				BinaryMapDataObject commonObj = commonMapData.get(id);
} else if (deleteId == -1 || !commonObj.containsType(deleteId)) {
```
If these ids are not correct and checks are not correct they need to be fixed instead of extra loops